src/library/commissioner_safe.cpp

```diff
@@ -91,20 +93,17 @@ const Config &CommissionerSafe::GetConfig() const
     return mImpl.GetConfig();
 }
 
-Error CommissionerSafe::Start()
+void CommissionerSafe::StartEventLoopThread()
 {
-    Error error = Error::kNone;
-
-    VerifyOrExit(!mEventThread.joinable(), error = Error::kAlready);
-
-    mEventThread = std::thread([this]() { IgnoreError(mImpl.Start()); });
+    VerifyOrDie(!mEventThread.joinable());
```

Comment:
We are always using `VerifyOrDie` instead of `Assert`. However, I think this is one of the situations that we should use `Assert` because it is a logical error, rather than some unrecoverable system error. Thought?

Reply:
agree.